tests/drivers/flash/api/page_layout/testcase.yaml

```diff
@@ -0,0 +1,43 @@
+common:
+  tags:
+    - drivers
+    - flash
+    - flash page layout
+  platform_allow:
+    - qemu_x86
+    - native_posix
+    - native_sim
+    - nrf52dk_nrf52832
+    - nucleo_f411re
+tests:
+  drivers.flash.api.page_layout.qemu_x86:
```

Comment:
This could be simplified a bit, simply have a test for userspace, and one without, then rely on the boards `.conf` files to include board specific configs, and omit the timeout, 60 seconds is already the default :) So the tests would look like
```
tests:
  drivers.flash.api.page_layout:
    platform allow:
      - qemu_x86
      - nrf52dk_nrf52832
      - nucleo_f411re
  drivers.flash.api.page_layout.userspace:
    extra_args:
      - OVERLAY_CONFIG=userspace.conf
    platform allow:
      - qemu_x86
      - nrf52dk_nrf52832
      - nucleo_f411re

```

Reply:
A small nit is that you can use
```
    extra_configs:
      - CONFIG_USERSPACE=y
```

instead of
```
OVERLAY_CONFIG=userspace.conf
```
To avoid the additional `userspace.conf` file